### PR TITLE
Allow easy deactivation of logging from code

### PR DESCRIPTION
**Pull request by lucatume**
This PR was born in the context of the work to port TEC and its tests to `tric` and GHA.
The PR adds a `disable` method to the logging Service Provider to allow, in tests and where it might be required, to disable logging completely.

The API is the following:

```php
// Enable logging.
tribe( 'log' )->enable();

// Disable logging.
tribe( 'log' )->disable();
```

### CODE REVIEW

File: src/Tribe/Log/Service_Provider.php [hunk 1]
```diff
@@ -22,6 +22,8 @@ class Service_Provider extends \tad_DI52_ServiceProvider {
 	 * @since 4.9.16
 	 */
 	public function register() {
+		$this->container->singleton( 'log', $this );
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:25]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: c781657fcd16f457e41592fe0c8765f2</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 2]
```diff
@@ -22,6 +22,8 @@ class Service_Provider extends \tad_DI52_ServiceProvider {
...
+		$this->container->singleton( static::class, $this );
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:26]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: c5bce384583a3ba28d7fc42021a6d36a</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 3]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
 
 		return $logging_engines;
 	}
+
+	/**
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:151]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: dcb92ce6070892036783bd78f76e5cda</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 4]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	 * Enables logging in the service provider, if not already enabled.
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:152]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 8b31288728b23558c1aba48ff410604f</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 5]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	 *
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:153]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 68858175c20032ebc8395d87f02bc6d6</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 6]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	 * @since TBD
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:154]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 1d021d76a4bce854a53ce5eb10356a56</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 7]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	 */
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:155]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: e6cb07bd2dbfd3256fb7a7340dbd47c1</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 8]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	public function enable(  ) {
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:156]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: Expected exactly one space before closing parenthesis; "  " found.
* :no_entry_sign: Expected 0 spaces between parenthesis of function declaration; 2 found
* :no_entry_sign: Opening brace should be on a new line

<sub>hash: c51cb8e1ef387e1bc7dbbfa748fdd6af</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 9]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+		if ( has_action( 'tribe_log', [ $this, 'dispatch_log' ] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:157]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 20ab11c59130cfa99de9b525999d444c</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 10]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+			return;
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:158]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 802ff1e85050027188bdce92bcf694c6</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 11]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+		}
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:159]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 482f0e561de21721e4fb20f8c0dec2dc</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 12]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+
+		remove_action( 'tribe_log', [ $this, 'dispatch_log' ] );
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:161]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 357ac1de965493906fde0b02ec45cb06</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 13]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	}
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:162]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 21dec10dc93bded00ee1220e3bf30b89</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 14]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+
+	/**
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:164]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 6e618498554b26aa21637f6bf2931b13</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 15]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	 * Disables the logging functions.
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:165]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 2415beb2a8d7333647bfec14ddd43b5e</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 16]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	 *
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:166]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 251439a8acaddaa2daa9134e76c4407d</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 17]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	 * @since TBD
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:167]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: c10f2ec527c98094521b16e35fa1bd7c</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 18]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	 */
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:168]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: be24d5c29010c58dfda28ab413fffd57</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 19]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	public function disable() {
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:169]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: Opening brace should be on a new line

<sub>hash: d66b682696da725e481ed0792f1d48c0</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 20]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+		if ( ! has_action( 'tribe_log', [ $this, 'dispatch_log' ] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:170]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: b6a831c07f802935a3dde54b42b9646e</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 21]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+			return;
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:171]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: cb93747c4a428b4fa23820701cd87449</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 22]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+		}
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:172]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 2e1c3c72bd43672f2dd77422429d674c</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 23]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+
+		remove_action( 'tribe_log', [ $this, 'dispatch_log' ] );
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:174]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 9edbbd1f7afc693ce471d5c830cae147</sub>

File: src/Tribe/Log/Service_Provider.php [hunk 24]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	}
```
Comment (tr1b0t) [inline src/Tribe/Log/Service_Provider.php:175]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: bfca363984d3a514ef82eff73ce342ca</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 28 error(s)

File: src/Tribe/Log/Service_Provider.php [hunk 25]
```diff
@@ -145,4 +147,30 @@ public function add_logging_engine( array $logging_engines = [] ) {
...
+	 * Enables logging in the service provider, if not already enabled.
+	 *
+	 * @since TBD
+	 */
+	public function enable() {
+		if ( has_action( 'tribe_log', [ $this, 'dispatch_log' ] ) ) {
+			return;
+		}
+
+		remove_action( 'tribe_log', [ $this, 'dispatch_log' ] );
```
Comment (paulmskim) [inline src/Tribe/Log/Service_Provider.php:161]: do you mean `add_action` here?

Comment (lucatume) [reply src/Tribe/Log/Service_Provider.php]: Yes, thanks.

**Review by paulmskim — APPROVED** [review]

**Pull request merged**